src/tcp_wrap.cc

```diff
@@ -211,8 +211,9 @@ void TCPWrap::Open(const FunctionCallbackInfo<Value>& args) {
                           args.Holder(),
                           args.GetReturnValue().Set(UV_EBADF));
   int fd = static_cast<int>(args[0]->IntegerValue());
-  uv_tcp_open(&wrap->handle_, fd);
+  int err = uv_tcp_open(&wrap->handle_, fd);
   wrap->set_fd(fd);
```

Comment:
Should this be `if (err == 0) wrap->set_fd(fd);`?

Reply:
It certainly could be. I'll update it. `PipeWrap()` does it this way, but it also throws from C++ instead of returning the error code.